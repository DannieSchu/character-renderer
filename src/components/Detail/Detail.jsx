import React from 'react';
import PropTypes from 'prop-types';
import Character from '../Characters/Character.jsx';

const Detail = ({ name, url, status, gender, born, description }) => {
  return (
    <>
      <Character name={name} url={url} /> 
      <p>Status: {status}</p>
      <p>Gender: {gender}</p>
      <p>Date of Birth: {born}</p>
      <p>{description}</p>
    </>
  );
};

Detail.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  born: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Detail;
