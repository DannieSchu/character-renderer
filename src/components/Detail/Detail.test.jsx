import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail.jsx';

describe('Detail', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Detail 
        name="John Fitzgerald Byers" 
        url="https://vignette.wikia.nocookie.net/x-files/images/a/a4/John_Fitzgerald_Byers.jpg/revision/latest?cb=20120717071453"
        status="Dead"
        gender="Male"
        born="22 November 1963"
        description="John Fitzgerald Byers was one of the three The Lone Gunmen (the others being Richard Langly and Melvin Frohike)." />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
