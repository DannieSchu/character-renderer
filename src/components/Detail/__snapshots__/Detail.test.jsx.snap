// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail matches a snapshot 1`] = `
<Detail
  born="22 November 1963"
  description="John Fitzgerald Byers was one of the three The Lone Gunmen (the others being Richard Langly and Melvin Frohike)."
  gender="Male"
  name="John Fitzgerald Byers"
  status="Dead"
  url="https://vignette.wikia.nocookie.net/x-files/images/a/a4/John_Fitzgerald_Byers.jpg/revision/latest?cb=20120717071453"
>
  <section>
    <Character
      name="John Fitzgerald Byers"
      url="https://vignette.wikia.nocookie.net/x-files/images/a/a4/John_Fitzgerald_Byers.jpg/revision/latest?cb=20120717071453"
    >
      <section>
        <h3>
          John Fitzgerald Byers
        </h3>
        <img
          alt="John Fitzgerald Byers"
          src="https://vignette.wikia.nocookie.net/x-files/images/a/a4/John_Fitzgerald_Byers.jpg/revision/latest?cb=20120717071453"
          width="250"
        />
      </section>
    </Character>
    <p>
      Status: 
      Dead
    </p>
    <p>
      Gender: 
      Male
    </p>
    <p>
      Date of Birth: 
      22 November 1963
    </p>
    <p>
      John Fitzgerald Byers was one of the three The Lone Gunmen (the others being Richard Langly and Melvin Frohike).
    </p>
  </section>
</Detail>
`;
